instrumentation/rocketmq-client/src/test/java/brave/rocketmq/client/ITRocketMQTracingTest.java

```diff
@@ -226,17 +230,17 @@ protected ConsumeOrderlyStatus handleMessage(MessageExt messageExt,
     consumer.subscribe(topic, "*");
     CountDownLatch latch = new CountDownLatch(1);
     AtomicReference<Span> reference = new AtomicReference<>();
-    consumer.registerMessageListener(new TracingMessageListenerOrderly(0, consumerTracing) {
+    MessageListenerOrderly messageListenerOrderly = consumerTracing.wrap(new MessageListenerOrderly() {
```

Comment:
I think this test can be simpler using similar style as JMS. For example, all you need to do is prove that you can read back the tag like so:

```java
  @Test void should_complete_on_callback() {
    producer.setAsync(new CompletionListener() {
      @Override public void onCompletion(Message message) {
        tracing.tracer().currentSpanCustomizer().tag("onCompletion", "");
      }

      @Override public void onException(Message message, Exception exception) {
        tracing.tracer().currentSpanCustomizer().tag("onException", "");
      }
    });
    producer.send(jms.queue, "foo");

    assertThat(testSpanHandler.takeRemoteSpan(PRODUCER).tags())
      .containsKeys("onCompletion");
  }
```

In other words, in this test and similar ones that are testing callbacks, you can do like

```java
 tracing.tracer().currentSpanCustomizer().tag("consumeMessage", "");
```

or whatever the callback method name is, and don't worry about countdown latches, because the IntegrationTestSpanHandler has a blocking queue and it will wait for the span to arrive.

Reply:
OK, I have already modified this part of the code.

Reply:
great. I like the amount of red in this diff!